app/validators/all_details_on_order_validator.rb

```diff
@@ -0,0 +1,39 @@
+class AllDetailsOnOrderValidator
+
+  cattr_accessor :order_detail_validator_class
+
+  # Build a validator that will take an Order in its constructor and validate each
+  # child order detail using the order_detail_validator_class.
+  #
+  # Example:
+  # validator = AllDetailsOnOrder.build(NotePresenceValidator)
+  # validator.new(order).valid?
+  #
+  # This will run NotePresenceValidator.new(order_detail).valid? for each
+  # element of @order.order_details
+  def self.build(order_detail_validator_class)
+    Class.new(AllDetailsOnOrderValidator) do
+      self.order_detail_validator_class = order_detail_validator_class
+    end
+  end
+
+  # This class should only be initialized through the `.build` method
+  def initialize(order)
+    raise "Should not be initialized directly. Use the `.build` factory method instead" unless order_detail_validator_class
+    @order = order
+  end
+
+  def valid?
+    # Use select instead of none? to make sure we loop over everything
```

Comment:
You're right. I switched from `select` to `reject`. The reason I added the comment was to explain why we don't compact this method to just ` @order.order_details.none? do |od| ...`. That, similar `any?`, will short circuit if it finds something in the list that matches.

Do you think the comment is useful if I update it to reflect the code?